### PR TITLE
Add the test package to coverage/inspection tests

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: benchmark/README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ You must be in the repo root directory to run these commands.
 Run the default benchmark suites:
 
 ```
-$ ./bench.sh --quick
+$ bench.sh --quick
```
Comment (adithyaov) [inline benchmark/README.md:81]: `bin/bench.sh`?
Is it symlinked?

File: default.nix [hunk 2]
```diff
@@ -90,11 +90,9 @@ let haskellPackages =
         # XXX On macOS cabal2nix does not seem to generate a dependency on
         # Cocoa framework.
         buildInputs =
-            [nixpkgs.haskellPackages.inspection-testing] ++
-            (if builtins.currentSystem == "x86_64-darwin"
-             then [nixpkgs.darwin.apple_sdk.frameworks.Cocoa]
-             else []
-            );
+            if builtins.currentSystem == "x86_64-darwin"
```
Comment (adithyaov) [inline default.nix:93]: Why remove this?

File: bin/build-lib.sh [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  then
+    echo $1
+  fi
+}
+
+# A group consisting of all known individual targets
+all_grp () {
+  { for i in $GROUP_TARGETS
+    do
+      eval "echo \$$i"
```
Comment (adithyaov) [inline bin/build-lib.sh]: Why can't you echo normally here?

File: bin/bench.sh [hunk 4]
```diff
@@ -413,6 +413,8 @@ run_reports() {
 # Execution starts here
 #-----------------------------------------------------------------------------
 
+cd $BENCH_SH_DIR/..
```
Comment (adithyaov) [inline bin/bench.sh]: Oh, I see. We are going up a directory.

File: benchmark/README.md (hunk 1 above)
Comment (adithyaov) [reply benchmark/README.md:81]: The previous statement says `You must be in the repo root directory to run these commands.`
You should be in the benchmarks directory?

**Review by adithyaov — APPROVED** [review]
LGTM. Just went through it. I might have missed something.

File: default.nix (hunk 2 above)
Comment (harendra-kumar) [reply default.nix:93]: It does not work as expected. We need `inspection-testing` to be installed as a ghc package in the shell which it does not.

File: bin/build-lib.sh (hunk 3 above)
Comment (harendra-kumar) [reply bin/build-lib.sh]: We have to echo the values of the variable names.

File: bin/test.sh [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  echo "Usage: $0 "
+  echo "       [--targets <"target1 target2 ..." | help>]"
+  echo "       [--slow]"
+  echo "       [--quick]"
+  echo "       [--cabal-build-options <option>]"
+  echo "       -- <hspec options or test names>"
+  echo
+  echo "--targets: targets to run, use 'help' for list of targets"
+  echo "--slow: take longer but run more tests"
+  echo "--quick: faster  but run fewer tests"
```
Comment (pranaysashank) [inline bin/test.sh:19]: is it fewer test suites or fewer number of quickcheck iterations ?

Comment (harendra-kumar) [reply bin/test.sh:19]: It is a to-do, does not anything right now. Just retained this from benchmarks script.

**Pull request merged**